### ISSUE TITLE
Upgrading from chart version 9.1.1 to 9.14.2 breaks install (missing CRD)

### ISSUE THREAD

**Issue by steveyackey** [traefik/traefik-helm-chart#359]
We recently realized we had been using https://containous.github.io/traefik-helm-chart instead of https://helm.traefik.io/traefik. We tried to upgrade, moving between these versions:

Previously installed chart   	Chart ver:9.1.1        	App ver:2.2.8      	
Newly upgraded chart            Chart ver:9.14.2       	App ver:2.4.2     

After upgrading, we received this error message in the traefik logs:
`1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: the server could not find the requested resource (get serverstransports.traefik.containo.us) `

When upgrading (even using helm upgrade --install), it seems that the serverstransports.traefik.containo.us was not being installed. With that error, all IngressRoutes stopped functioning. I was able to fix the upgrade by manually installing the CRD (just doing a kubectl apply with the CRD manifest from this repo). 

The workaround seems to fix things, but wanted to report the issue in case others faced it as well.

**Comment by waterdrop01** [comment]
As per Steve feedback, here is the workaround:
```bash
curl https://raw.githubusercontent.com/traefik/traefik-helm-chart/master/traefik/crds/serverstransports.yaml | kubectl apply -f -
helm upgrade traefik traefik/traefik --values helm-traefik-values.yml
```

**Comment by FedeBev** [comment]
Hi there,

same issue here, the suggested workaround did the trick for me.
However, my case is a bit different: I've only upgraded the charts version from 9.11.0 to 9.14.3, always using the same repo (https://helm.traefik.io/traefik)

**Comment by lion7** [comment]
```
The workaround seems to fix things, but wanted to report the issue in case others faced it as well.
```
Thanks for that, just faced the same problem when moving from the old URL and this workaround solved it 😄

**Comment by adhodgson1** [comment]
I saw this issue this morning, another workaround is to uninstall the chart and reinstall it from scratch, the CRDs are then created. I am wondering if this may be a Helm issue, we are on Helm 3.4.1.

**Comment by adhodgson1** [comment]
Doing some more tests on this I think this behaviour is by design:
https://helm.sh/docs/chart_best_practices/custom_resource_definitions/
"There is no support at this time for upgrading or deleting CRDs using Helm. This was an explicit decision after much community discussion due to the danger for unintentional data loss. Furthermore, there is currently no community consensus around how to handle CRDs and their lifecycle. As this evolves, Helm will add support for those use cases."
The first chart version with this new CRD in place was 9.13.0. In my view the bigger question is what is the best way for Traefik to handle the CRD updates during install? I know some charts handle CRDs as separate steps.

**Comment by sandroden** [comment]
It solved the problem for me as well even though I'm a little puzzled as it seems I have 9.1.1 not 9.14.2:

``` # microk8s.helm3  list -n traefik -a
NAME    NAMESPACE       REVISION        UPDATED                                         STATUS          CHART           APP VERSION
traefik traefik         2               2021-06-07 13:00:33.800609917 +0200 CEST        deployed        traefik-9.1.1   2.2.8
```
I'm using version 3.0.2

**Comment by githubixx** [comment]
The same happened for me again while upgrading Helm chart from `10.1.1` to `10.3.0` and `image.tag` from `2.4.13` to `2.5.0`. After upgrading I saw this error in the logs:

```
E0818 16:43:31.369806       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.0/tools/cache/reflector.go:167: Failed to watch *v1alpha1.MiddlewareTCP: failed to list *v1alpha1.MiddlewareTCP: the server could not find the requested resource (get middlewaretcps.traefik.containo.us)
```

This also caused all sites to be not available anymore as the browsers complained that the site uses a "self signed certificate". I did a `helm uninstall ...` to remove everything and installed everything again. After that everything worked again.

I suspect installing `https://raw.githubusercontent.com/traefik/traefik-helm-chart/master/traefik/crds/middlewarestcp.yaml` first would also have helped but this is nothing you know upfront.

So is there something planed to fix this to make upgrades via helm more convenient?
